doc: clarify supported upgrade and rollback scenarios

lmb opened the pull request:
Make it explicit in the upgrade documentation that the only upgrade and rollback combination we test are the latest patches of consecutive minor releases.

```release-note
Document supported upgrade and rollback paths
```

learnitall commented on Documentation/operations/upgrade.rst:
```diff
@@ -140,9 +140,8 @@ Step 1: Upgrade to latest patch version
-Cilium release series first. The latest patch releases for each supported
-version of Cilium are `here <https://github.com/cilium/cilium#stable-releases>`_.
+1.x to 1.y, it is recommended to upgrade to the `latest patch release
+<https://github.com/cilium/cilium#stable-releases>` for a Cilium release series first.
```
Missing double underscore after link. See https://docs.cilium.io/en/stable/contributing/docs/docsstyle/#links.

```suggestion
<https://github.com/cilium/cilium#stable-releases>`__ for a Cilium release series first.
```

learnitall commented on Documentation/operations/upgrade-warning.rst:
```diff
@@ -7,8 +7,12 @@
 
-   Do not upgrade to \ |NEXT_RELEASE|.0 before reading the section
-   :ref:`current_release_required_changes` and completing the required steps.
-   Skipping this step may lead to an non-functional upgrade.
```
IMO I think the wording in the paragraph you removed is helpful to users because it gives a very clear warning with instructions. Users may interpret what it means to "pay particular attention" differently, but it's harder to misinterpret "do not upgrade before reading". Would you mind reverting this change? What do you think?

learnitall commented on Documentation/operations/upgrade-warning.rst:
```diff
@@ -7,8 +7,12 @@
-   Skipping this step may lead to an non-functional upgrade.
+   The only tested rollback path is between consecutive minor releases. This means
+   that going from (a hypothethical) 1.42 to 1.43 and back is supported while going
+   to 1.44 is not.
```
Couple of suggestions here. Let me know what you think:

* This statement is also relevant to upgrade paths
* Use smaller version numbers, as these are commonly associated with examples
* Provide a clear instruction to users given this information

```suggestion
   The only tested rollback and upgrade path is between consecutive minor releases.
   Always perform rollbacks and upgrades between one minor release at a time.
   This means that going from (a hypothetical) 1.1 to 1.2 and back is supported
   while going from 1.1 to 1.3 and back is not.
```

learnitall reviewed (changes requested):
Good idea adding this, thank you! I have a couple of suggestions for you to help make these instructions explicit to users, let me know what you think.

lmb committed:
doc: clarify supported upgrade and rollback scenarios

Make it explicit in the upgrade documentation that the only upgrade
and rollback combination we test are the latest patches of consecutive
minor releases.

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

learnitall reviewed (approved):
Thanks!
